spec/services/support_interface/tad_degree_class_export_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+RSpec.describe SupportInterface::TADDegreeClassExport do
+  describe '#call' do
+    it 'correctly breaks down subject choice by route' do
+      drama = create(:subject, code: '13')
+      first_application_form = create(:completed_application_form)
+      create(:application_qualification, level: 'degree', grade: 'Upper second-class honours (2:1)', application_form: first_application_form)
+      scitt_provider = create(:provider, provider_type: 'scitt')
+      first_course = create(:course, provider: scitt_provider, subjects: [drama])
+      first_course_option = create(:course_option, course: first_course)
+
+      create(:application_choice, :with_declined_offer, course_option: first_course_option, application_form: first_application_form)
+
+      second_application_form = create(:completed_application_form)
+      create(:application_qualification, level: 'degree', grade: 'Upper second-class honours (2:1)', application_form: second_application_form)
+      lead_school_provider = create(:provider, provider_type: 'lead_school')
+      second_course = create(:course, provider: lead_school_provider, subjects: [drama])
+      second_course_option = create(:course_option, course: second_course)
+
+      create(:application_choice, :with_declined_offer, course_option: second_course_option, application_form: second_application_form)
+
+      data = described_class.new.call
+
+      expect(data).to include(
+        {
+          subject: :drama,
+          route: 'scitt',
+          degree_class: 'Upper second-class honours (2:1)',
+          applications: 1,
+          offers_received: 0,
+          number_of_acceptances: 0,
+          number_of_declined_applications: 1,
+          number_of_rejected_applications: 0,
+          number_of_withdrawn_applications: 0,
+        },
+        {
+          subject: :drama,
+          route: 'lead_school',
+          degree_class: 'Upper second-class honours (2:1)',
```

Comment:
Just wondering what the requirement is here about how to format these values. Are they the raw data values, e.g. `lead_school` or are they humanised to `Lead school`?

Reply:
they are happy for it to remain as is